### PR TITLE
[4.4] Fixing php compatibility issues 4

### PR DESCRIPTION
**Pull request by Hackwar**
### Summary of Changes
This is one of several PRs to separate out the fixes from #38471. The `track_errors` feature was deprectaed in PHP 7.2 and removed in PHP 8.0. Most places where we use this, it isn't actually used.

### Testing Instructions
Codereview

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: libraries/src/Http/Transport/SocketTransport.php [hunk 1]
```diff
@@ -229,30 +229,14 @@ protected function connect(UriInterface $uri, $timeout = null)
...
-            if (!$php_errormsg) {
-                // Error but nothing from php? Create our own
-                $php_errormsg = sprintf('Could not connect to resource %s: %s (error code %d)', $uri, $err, $errno);
-            }
-
-            // Restore error tracking to give control to the exception handler
-            ini_set('track_errors', $track_errors);
-
-            throw new \RuntimeException($php_errormsg);
+            throw new \RuntimeException(sprintf('Could not connect to resource %s: %s (error code %d)', $uri, $err, $errno));
```
Comment (HLeithner) [inline libraries/src/Http/Transport/SocketTransport.php:237]: shouldn't this check `error_get_last()` for an error message?

File: libraries/src/Http/Transport/StreamTransport.php [hunk 2]
```diff
@@ -142,29 +142,13 @@ public function request($method, UriInterface $uri, $data = null, array $headers
...
-            if (!$php_errormsg) {
-                // Error but nothing from php? Create our own
-                $php_errormsg = sprintf('Could not connect to resource: %s', $uri);
-            }
-
-            // Restore error tracking to give control to the exception handler
-            ini_set('track_errors', $track_errors);
-
-            throw new \RuntimeException($php_errormsg);
+            throw new \RuntimeException(sprintf('Could not connect to resource: %s', $uri));
```
Comment (HLeithner) [inline libraries/src/Http/Transport/StreamTransport.php:149]: please check `error_get_last()` as in the original code

**Comment by HLeithner** [comment]
and please rebase on 4.4

File: libraries/src/Http/Transport/SocketTransport.php (hunk 1 above)
Comment (MacJoom) [reply libraries/src/Http/Transport/SocketTransport.php:237]: @Hackwar please review your code - thanks

File: libraries/src/Http/Transport/StreamTransport.php (hunk 2 above)
Comment (MacJoom) [reply libraries/src/Http/Transport/StreamTransport.php:149]: @Hackwar please review your code - thanks

**Comment by Hackwar** [comment]
This PR is half a year old and not that important to me that I want to push this further. Closing it.

**Pull request closed without merging**